src/RateLimit/Entity/RateLimitController.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\restful\RateLimit\Entity\RateLimitController.
+ */
+
+namespace Drupal\restful\RateLimit\Entity;
+
+class RateLimitController extends \EntityAPIController {}
```

Comment:
Having your own typed objects is usually a good idea. This means that you have a way to check if that controller is a rate limit controller. You also give yourself some room to add new methods to it without having to change the API and breaking compat with the major version.

This carries from 1.x.